### ISSUE TITLE
Nullable reference types: string.IsNullOrEmpty / IsNullOrWhitespace not recognized

### ISSUE THREAD

**Issue by bkoelman** [dotnet/roslyn#9977]
I have been playing with nullable reference types, which is already working remarkably great! To try it out, I have [patched up](https://github.com/bkoelman/ResharperCodeContractNullability/tree/staticnull) my analyzer to convert Resharper's `(Item)NotNull/CanBeNullAttribute` annotations to nullable reference types. Then ran that on an [actual annotated codebase](https://github.com/bkoelman/DogAgilityCompetitionManagement). 

During that process I discovered some oddities, which I'll create issues for here.

**Version Used**: Built from source, branch NullableReferenceTypes (commit https://github.com/dotnet/roslyn/commit/8553cdac75c2b0901e8ae401aaa23022f0689935)

**Steps to Reproduce**:
The next block of code raises warning 8202: Possible dereference of a null reference:

``` csharp
public class Test
{
    public static void M(string? text)
    {
        if (string.IsNullOrWhiteSpace(text))
        {
            return;
        }
        string trimmed = text.Trim(); // CS8202
    }
}
```

**Expected Behavior**: No CS8202 warning, because `string.IsNullOrWhiteSpace` (and `string.IsNullOrEmpty`) returns `true` if `text` is `null`, which makes the `text.Trim()` line unreachable. 

I believe these two methods are so commonly used that it's worth special-casing them.

**Actual Behavior**:
CS8202 warning on the `text` after the if statement.

@AlekseyTs

**Comment by jcouv** [comment]
[Confirmed](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhAEssmATEAagB8AAgGYABIIBMo6ihgBYAFABvRaLXixggIwA2cQBZRAWQAU2gAwB+UTAgAPGAEpV6lQvUfRbMKLNbzAHQAksiYOADyCADqABZstgDK9FBUJrYOjs7unmpuOTmCAOwA3C45AL5lnhY2CGwAtvUQXKIAvDb2MAHUdfUmjsWiAPRDogDCCQAcEuYSVZUK5UA===) this was fixed.

A very small set of APIs are recognized at the moment (`Debug.Assert`, `IsNullOrEmpty`, ...). We'll likely have a more general solution.
I'll go ahead and close this issue. Thanks

**Comment by qidydl** [comment]
Has there been a regression with this lately? I'm seeing this same problem in Visual Studio 2019 16.9.0 / .NET SDK 5.0.200:

![image](https://user-images.githubusercontent.com/5160192/110004097-0d741b00-7ce5-11eb-8428-e8b1f4b93c49.png)
